nntrainer/tensor/manager.cpp

```diff
@@ -174,22 +174,12 @@ void Manager::trackWeights(std::vector<Weight> &ws) {
   max_grad_size = std::max(max_grad_size, grad_size);
 }
 
-/**
- * @brief Allocate and initialize the weight variable
- */
-void Manager::initialize() {
-  if (total_weight_size == 0) {
-    ml_logw("Nothing done on initialize because there is no weight registered");
-    return;
-  }
-  using AllocFunc = std::function<Tensor(const TensorDim &, size_t)>;
-
+Manager::AllocFunc Manager::getAllocFunc(bool is_weight) {
```

Comment:
I like this :)
Could you reflect https://github.com/nnstreamer/nntrainer/pull/859/commits/e4f66430754e4b5e5366a714512ccf43fe26a398 here?
I think the commit suits here.